agent/consul/leader_connect_ca_test.go

```diff
@@ -85,26 +85,27 @@ func (m *mockCAServerDelegate) ApplyCARequest(req *structs.CARequest) (interface
 	switch req.Op {
 	case structs.CAOpSetConfig:
 		if req.Config.ModifyIndex != 0 {
-			act, err := m.store.CACheckAndSetConfig(idx+1, req.Config.ModifyIndex, req.Config)
+			var emptyResp interface{}
+			err := m.store.CACheckAndSetConfig(idx+1, req.Config.ModifyIndex, req.Config)
 			if err != nil {
 				return nil, err
 			}
 
-			return act, nil
+			return emptyResp, nil
```

Comment:
This should mirror `FSM.applyConnectCAOperation` so that our tests match production (see the godoc comment I added to this method).

So I think this one should be simply:
```go
return c.state.CASetConfig(idx, req.Config)
```

and probably the cases below should also be updated to match `FSM.applyConnectCAOperation`